### PR TITLE
docs: add Daft integration

### PR DESCRIPTION
**Pull request by avriiil**
This adds an integration page for using Delta Lake with Daft.

### CODE REVIEW

**Comment by ion-elgreco** [comment]
Not so sure about promoting daft here, they collect telemetry by default...

**Comment by MrPowers** [comment]
Seems like their telemetry gathering is responsible, but open to thoughts: https://www.getdaft.io/projects/docs/en/latest/faq/telemetry.html

**Comment by ion-elgreco** [comment]
I don't really agree, no one is going to know telemetry is on when you pip install it.

They likely also violate laws and regulations for users falling under GDPR regulation

**Comment by MrPowers** [comment]
@ion-elgreco - I don't think that non-identifiable data falls under the scope of GDPR.

**Comment by jaychia** [comment]
Hey folks! I'm one of the maintainers of the Daft project, just chiming in here to help clarify some of the work in Daft.

All telemetry collected is non-identifiable, which should not fall in scope of GDPR.

We also noticed that other frameworks which delta-rs already integrates with and currently includes in its documentation also do collect telemetry (though maybe they are not as public as we are so it is less discoverable).

* Ray: https://docs.ray.io/en/latest/cluster/usage-stats.html
* Dagster: https://docs.dagster.io/about/telemetry

I think that this is important to call out because I wanted to keep discussions on this PR related to the integrations with Daft, and perhaps if the Delta project needs to discuss its stance on its integrations' telemetry policies that can be discussed in a separate thread/issue?

Happy to provide any more clarifications if necessary!

**Comment by avriiil** [comment]
Hey @ion-elgreco @MrPowers - curious on how we can best proceed here? Would be great to get the integration docs in so that folks who read [this new Delta blog](https://delta.io/blog/daft-delta-lake-integration/) can access the relevant documentation :)

**Comment by ion-elgreco** [comment]
I would rather see a mention on all these integrations that they collect telemetry which can be opted out from, just a small note at the bottom..

**Comment by avriiil** [comment]
@ion-elgreco - that seems fine to me. Perhaps out of scope for this PR since it will affect the docs for the other integration techs too?

**Comment by avriiil** [comment]
@ion-elgreco I've gone ahead and added a note at the end. LMK what you think :)

File: docs/integrations/delta-lake-daft.md [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+1. **Large datasets** that don't fit into memory or would benefit from parallelization
+2. **Multimodal data types**, such as images, JSON, vector embeddings, and tensors
+3. **ML workloads** that would benefit from interactive computation within DataFrame (via UDFs)
+
+Take a look at the [Daft tutorials](https://www.getdaft.io/projects/docs/en/latest/user_guide/tutorials.html) for in-depth examples of each use case.
+
+## Contribute to `daft`
+
+Excited about Daft and want to contribute? Join us on [Github](https://github.com/Eventual-Inc/Daft) 🚀
```
Comment (ion-elgreco) [inline docs/integrations/delta-lake-daft.md:348]:
```suggestion
Excited about Daft and want to contribute? Join them on [Github](https://github.com/Eventual-Inc/Daft) 🚀
```

Comment (ion-elgreco) [reply docs/integrations/delta-lake-daft.md:348]: @avriiil if you can resolve this one then it's good to go

**Review by ion-elgreco — APPROVED** [review]

**Comment by avriiil** [comment]
Thanks for your help getting this through @ion-elgreco!

**Pull request merged**